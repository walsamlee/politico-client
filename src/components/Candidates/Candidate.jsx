/* eslint-disable react/prop-types */
import React, { Component } from 'react';
import Vote from '../containers/Vote';

class Candidate extends Component {
  dosomething = () => {
    const a = 4;
    return a;
  }

  render() {
    const { details } = this.props;

    const voter = parseInt(localStorage.getItem('whom'), 10);

    const voteData = {
      office: details.officeid,
      candidate: details.userid,
      voter,
    };

    return (
      <div className="vote">
        <Vote voteData={voteData} />
        <div className="candidate-details">
          <div className="candidate-image">
            <img src={details.passporturl} alt="candidate" />
          </div>
          <div className="candidate-particulars">
            <div className="candidate-name">
              <p>
                {details.firstname}
            &nbsp;
                {details.lastname}
              </p>
            </div>
            <div className="candidate-office">
              <h4>{details.name}</h4>
            </div>
          </div>
        </div>
        <div className="party-details">
          <div className="party-details-logo">
            <img src={details.logourl} alt="Party Logo" />
          </div>
        </div>
      </div>
    );
  }
}

export default Candidate;
