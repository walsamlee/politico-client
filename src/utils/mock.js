const voteData = {
  office: 1,
  candidate: 1,
  voter: 1,
};

const data = {
  candidate: {
    candidates: [
      {
        officeid: 4,
        firstname: 'Admin',
        lastname: 'Admin',
        othername: 'Sam',
        passporturl: 'https://res.cloudinary.com/walsam/image/upload/v1559164438/IMG_5704.jpg',
        userid: 1,
        partyid: 9,
        name: 'Legislative',
        logourl: 'https://res.cloudinary.com/walsam/image/upload/v1550613166/difd2f30syyz5fynogsz.png',
      },
      {
        officeid: 6,
        firstname: 'Hassan',
        lastname: 'Hadji',
        othername: 'Chikaso',
        passporturl: '',
        userid: 2,
        partyid: 8,
        name: 'Chairman',
        logourl: 'https://res.cloudinary.com/walsam/image/upload/v1550612284/zausu6uc1iet2lo4fsqr.png',
      },
    ],
  },
  getOffices: {
    offices: [],
  },
  parties: {
    parties: [],
  },
};

const details = {
  office: 1,
  candidate: 2,
};

const token = '';

const login = {
  login: {
    token: '',
    user: [
      {
        user: {
          isAdmin: '',
          id: 1,
          passportUrl: '',
          email: '',
          name: '',
          phoneNumber: 14141414441,
        },
      },
    ],
  },
};

const partiesData = {
  parties: {
    fetching: '',
    fetched: '',
    parties: [
      {
        logoUrl: 'https://res.cloudinary.com/walsam/image/upload/v1550613166/difd2f30syyz5fynogsz.png',
        name: 'wrrryery',
        id: 1,
        fetching: true,
      },
      {
        logoUrl: 'https://res.cloudinary.com/walsam/image/upload/v1550613166/difd2f30syyz5fynogsz.png',
        name: 'uiyt',
        id: 2,
        fetching: true,
      },
    ],
  },
};

const partyData = {
  id: '',
  image: '',
  partyname: '',
  parthqaddress: '',
};

const profileData = {
  picture: '',
  name: '',
  email: '',
  phone: '',
};

const registerData = {
  preview: '',
  firstname: '',
  lastname: '',
  othername: '',
  telephone: '',
  email: '',
  password: '',
};

const candidate = {
  passporturl: '',
  firstname: '',
  lastname: '',
  name: '',
  logourl: '',
};

const party = {
  id: '',
  name: '',
};

const aspirantData = {
  id: '',
  office: '',
  party: '',
};

const setPartyState = [
  {
    id: 1,
    logoUrl: '',
    name: '',
    hqAddress: '',
  },
];

const mock = {
  voteData,
  data,
  details,
  token,
  login,
  partiesData,
  partyData,
  profileData,
  registerData,
  candidate,
  party,
  aspirantData,
  setPartyState,
};

export default mock;
