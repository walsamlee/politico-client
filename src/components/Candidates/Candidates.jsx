/* eslint-disable react/prop-types */
/* eslint-disable jsx-a11y/label-has-for */
/* eslint-disable jsx-a11y/label-has-associated-control */
import React, { Component } from 'react';
import Candidate from './Candidate';

class Candidates extends Component {
  componentDidMount() {
    const token = localStorage.getItem('token');
    const { fetchCandidates } = this.props;
    fetchCandidates(token);
  }

  render() {
    const { data } = this.props;
    const { candidate } = data;
    const { candidates } = candidate;

    return (
      <div className="dashboard">
        <div className="history-board">
          <div className="candidate-form">
            <form action="">
              <label htmlFor="office">Choose Office: </label>
              <select name="office" id="office">
                {candidates.map((office) => {
                  const { name } = office;
                  return <option value={name}>{name}</option>;
                })}
              </select>
            </form>
          </div>
          {
            candidates.map(item => (
              <Candidate key={item.officeid} details={item} />
            ))
          }
        </div>
      </div>
    );
  }
}

export default Candidates;
