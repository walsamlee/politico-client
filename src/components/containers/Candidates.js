import { connect } from 'react-redux';
import Candidates from '../Candidates/Candidates';
import fetchCandidates from '../../actions/candidateAction';

const mapStateToProps = state => ({
  data: state,
});

const matchDispatchToProps = dispatch => ({
  fetchCandidates: (token) => {
    dispatch(fetchCandidates(token));
  },
});

const CandidatesContainer = connect(
  mapStateToProps,
  matchDispatchToProps,
)(Candidates);

export default CandidatesContainer;
