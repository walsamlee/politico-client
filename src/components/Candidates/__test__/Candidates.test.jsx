import React from 'react';
import { shallow } from 'enzyme';
import mock from '../../../utils/mock';
import Candidates from '../Candidates';

const { data } = mock;

describe('<Candidates />', () => {
  it('should render candidates component', () => {
    const wrapper = shallow(
      <Candidates data={data} fetchCandidates={jest.fn()} />,
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div'));
  });
});
