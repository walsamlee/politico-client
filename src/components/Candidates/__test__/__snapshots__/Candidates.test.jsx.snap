// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Candidates /> should render candidates component 1`] = `ShallowWrapper {}`;
